src/main/java/io/gravitee/el/spel/context/SpelTemplateContext.java

```diff
@@ -49,4 +77,47 @@ public Object lookupVariable(String name) {
     public EvaluationContext getContext() {
         return context;
     }
+
+    public Single<EvaluationContext> evaluationContext(CachedExpression expression) {
+        if (deferredVariables != null) {
+            final List<Completable> collect = deferredVariables
```

Comment:
In order to avoid browsing the map to build a list and then browse the list I would suggest to replace the stream+merge with

```suggestion
         return Flowable.fromIterable(deferredVariables.entrySet())
                    .filter(deferredEntry -> requiresDeferredVariable(expression, deferredEntry))
                    .flatMapCompletable(deferredEntry -> resolveDeferred(deferredEntry.getKey(), deferredEntry.getValue()))
                    .andThen(Single.just(context));
```